src/SAML2/Message.php

```diff
@@ -603,8 +602,9 @@ public function getExtensions()
     public function setExtensions($extensions)
     {
         assert(is_array($extensions) || is_null($extensions));
-
-        $this->extensions = $extensions;
+        if (is_array($extensions)) {
```

Comment:
I'm not sure `$this->extensions` is ever evaluated for `null` in the code (e.g.: in order to simply ignore it), but if it does, the lack of an `is_array()` check was effectively allowing us to clear the property.